### PR TITLE
Orca: update NCF examples to support training with ml-100k dataset

### PR DESCRIPTION
**Pull request by zpeng1898**
put datasets `ml-1m` and `ml-100k` in the directory such as `/path/to/NCF`

run with `ml-1m` dataset with `local` mode:
`python pytorch_train_spark_dataframe.py --data_dir /path/to/NCF`

run with `ml-100k` dataset with `local` mode
`python pytorch_train_spark_dataframe.py --data_dir /path/to/NCF --dataset ml-100k`

run with `ml-100k` dataset with `yarn-client` mode
`python pytorch_train_spark_dataframe.py --data_dir /path/to/NCF --dataset ml-100k --cluster_mode yarn-client`

You can replace `.py` file with `pytorch_train_xshards.py`, ` pytorch_train_dataloader.py`, ` tf_train_spark_dataframe.py`, ` tf_train_xshards.py`

### CODE REVIEW

**Comment by hkvision** [comment]
u.item     -- Information about the items (movies); this is a tab separated
              list of
              movie id | movie title | release date | video release date |
              IMDb URL | unknown | Action | Adventure | Animation |
              Children's | Comedy | Crime | Documentary | Drama | Fantasy |
              Film-Noir | Horror | Musical | Mystery | Romance | Sci-Fi |
              Thriller | War | Western |
              The last 19 fields are the genres, a 1 indicates the movie
              is of that genre, a 0 indicates it is not; movies can be in
              several genres at once.
              The movie ids are the ones used in the u.data data set.

File: python/orca/tutorial/NCF/utils.py [hunk 1]
```diff
@@ -26,8 +26,10 @@
 
 def parse_args(description, mode="train"):
     parser = argparse.ArgumentParser(description=description)
-    parser.add_argument("--data_dir", type=str, default="./ml-1m",
+    parser.add_argument("--data_dir", type=str, default="./",
                         help="The path to load data from local or remote resources.")
+    parser.add_argument("--dataset", type=str, default="ml-1m",
```
Comment (hkvision) [inline python/orca/tutorial/NCF/utils.py:31]: make this a choice?
image

File: python/orca/tutorial/NCF/process_spark_dataframe.py [hunk 2]
```diff
@@ -158,7 +187,7 @@ def get_label_cols():
 
     sc = init_orca_context()
```
Comment (hkvision) [inline python/orca/tutorial/NCF/process_spark_dataframe.py:188]: change this to init_orca? also for other data files?

File: python/orca/tutorial/NCF/process_xshards.py [hunk 3]
```diff
@@ -68,22 +68,48 @@ def split_dataset(df):
...
+        items = read_csv(
+            os.path.join(data_dir, dataset, "u.item"),
+            sep="|", header=None,
+            names=["item"]+[f"col{i}" for i in range(19)],
+            usecols=[0]+list(range(5, 24)),
+            dtype=str)
+
+        # merge multiple one-hot columns into one category column
+        def merge_one_hot_cols(df):
+            df["item"] = df["item"].astype(np.int64)
```
Comment (hkvision) [inline python/orca/tutorial/NCF/process_xshards.py:107]: why add this?

File: python/orca/tutorial/NCF/process_xshards.py [hunk 4]
```diff
@@ -68,22 +68,48 @@ def split_dataset(df):
...
+            category_cols = df.iloc[:, 1:]
+            df["category"] = pd.Series([''.join(row) for row in category_cols.values])
```
Comment (hkvision) [inline python/orca/tutorial/NCF/process_xshards.py:109]: is it efficient?
What about something like`df["Period"] = df[["Courses", "Duration"]].apply(lambda x: "-".join(x), axis =1)`?

File: python/orca/tutorial/NCF/process_xshards.py (hunk 3 above)
Comment (zpeng1898) [reply python/orca/tutorial/NCF/process_xshards.py:107]: image
The datatype of all columns is str when read items using `read_csv`. So the column `item` should be set to `int64` manually.

Comment (hkvision) [reply python/orca/tutorial/NCF/process_xshards.py:107]: okay, then what about using np.int32?

Comment (hkvision) [reply python/orca/tutorial/NCF/process_xshards.py:107]: or as you need to concat them to string, does it matter whether it is a str or int?

File: python/orca/tutorial/NCF/pytorch_dataset.py [hunk 5]
```diff
@@ -90,35 +90,54 @@ def __getitem__(self, idx):
...
+            users = pd.read_csv(
+                os.path.join(local_dir, "u.user"),
+                sep="|", header=None, names=["user", "age", "gender", "occupation", "zipcode"],
+                usecols=[0, 1, 2, 3, 4],
+                dtype={0: np.int64, 1: np.int32, 2: str, 3: str, 4: str})
+            items = pd.read_csv(
+                os.path.join(local_dir, "u.item"),
+                sep="|", header=None, names=["item"]+[f"col{i}" for i in range(19)],
+                usecols=[0]+list(range(5, 24)),
+                dtype=str, encoding="latin-1")
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_dataset.py:128]: dtype=int64?

File: python/orca/tutorial/NCF/pytorch_dataset.py [hunk 6]
```diff
@@ -90,35 +90,54 @@ def __getitem__(self, idx):
...
+
+            # merge multiple one-hot columns into one category column
+            items["item"] = items["item"].astype(np.int64)
+            category_cols = items.iloc[:, 1:]
+            items["category"] = pd.Series([''.join(row) for row in category_cols.values])
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_dataset.py:133]: also change this using apply?

File: python/orca/tutorial/NCF/process_xshards.py (hunk 4 above)
Comment (hkvision) [reply python/orca/tutorial/NCF/process_xshards.py]: efficiency may need to be further verified. At least for easier to understand, using apply may be better~

**Pull request merged**